lib/systems/include/MainChargeSystem.h

```diff
@@ -0,0 +1,18 @@
+#ifndef __MAINCHARGE_H__
+#define __MAINCHARGE_H__
+
+
+#include "CCUPID.h"
+#include "EthernetInterface.h"
+#include "CCUParams.h"
+
+
```

Comment:
Before starting to write code, make sure you're very clear on _what_  this class is responsible for doing. With any object oriented design (OOD), you need to define a few things:
- what does this class *have*
- what does this class *do*

from those, you can define
- what does this class *need* in order to *do* those things

then, and only then, should you start writing code.